pkg/pipelinerunmetrics/metrics.go

```diff
@@ -396,9 +414,37 @@ func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
 	var runningPipelineRuns int
 	var trsWaitResolvingTaskRef int
 	var prsWaitResolvingPipelineRef int
+	countMap := map[string]int{}
 
 	for _, pr := range prs {
+		pipelineName := getPipelineTagName(pr)
+		pipelineRunKey := ""
+		mutators := []tag.Mutator{
+			tag.Insert(namespaceTag, pr.Namespace),
+			tag.Insert(pipelineTag, pipelineName),
+		}
+		if r.cfg != nil {
+			switch r.cfg.RunningPipelinerunLevel {
+			case runningPRLevelPipelinerun:
+				pipelineRunKey = pipelineRunKey + "#" + pr.Name
+				fallthrough
+			case runningPRLevelPipeline:
+				pipelineRunKey = pipelineRunKey + "#" + pipelineName
+				fallthrough
+			case runningPRLevelNamespace:
+				pipelineRunKey = pipelineRunKey + "#" + pr.Namespace
+			case runningPRLevelBlank:
+			default:
+				return fmt.Errorf("RunningPipelineRunLevel value \"%s\" is not valid ", r.cfg.RunningPipelinerunLevel)
+			}
+		}
+		ctx_, err_ := tag.New(context.Background(), mutators...)
+		if err_ != nil {
+			return err
+		}
 		if !pr.IsDone() {
+			countMap[pipelineRunKey]++
+			metrics.Record(ctx_, runningPRs.M(float64(countMap[pipelineRunKey])))
```

Comment:
The `countMap` is only complete at the end of the loop, so why is the metric reported here?

Reply:
The metric might have different tags. For instance, in case we configure the metric to be at "namespace" level, it will have one counter for each namespace. How does the metric API know what tag value it is that we're counting here? Shouldn't we pass the `pipelineRunKey` somehow as well?

Reply:
> The metric might have different tags. For instance, in case we configure the metric to be at "namespace" level, it will have one counter for each namespace. How does the metric API know what tag value it is that we're counting here? Shouldn't we pass the `pipelineRunKey` somehow as well?

pipelineRunKey is generated on the basis of configuration  if config is at namespace level then namespace is the key.

Reply:
> The `countMap` is only complete at the end of the loop, so why is the metric reported here?

There is no way of knowing if  there is any more pipeline runs for same key. 

Incase there are more metrics for same key then on next instance metric is reported with updated value and we are just considering the last value of the metric so same is updated accordingly.